### PR TITLE
Remove Sleep typeclass

### LINKED ISSUE #525: Remove the Sleep type class
Use Temporal instead

### PR DESCRIPTION
**Pull request by AlejandroBudy**
fixes #525

### CODE REVIEW

File: modules/docs/src/main/mdoc/docs/combinators.md [hunk 1]
```diff
@@ -10,38 +10,38 @@ retries.
...
-arbitrary `Monad` that is not a `MonadError`.  Your operation doesn't throw
+with the result or you want to retry. It is useful when you are working in an
+arbitrary `Monad` that is not a `MonadError`. Your operation doesn't throw
 errors, but you want to retry until it returns a value that you are happy with.
 
 The API (modulo some type-inference trickery) looks like this:
 
 ```scala
-def retryingOnFailures[M[_]: Monad: Sleep, A](policy: RetryPolicy[M],
+def retryingOnFailures[M[_]: Monad: Temporal, A](policy: RetryPolicy[M],
```
Comment (AlejandroBudy) [inline modules/docs/src/main/mdoc/docs/combinators.md:33]:
```suggestion
def retryingOnFailures[M[_]: Temporal, A](policy: RetryPolicy[M],
```

File: modules/docs/src/main/mdoc/docs/mtl-combinators.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ whether a given error is worth retrying.
 The API (modulo some type-inference trickery) looks like this:
 
 ```scala
-def retryingOnSomeErrors[M[_]: Monad: Sleep, A, E: Handle[M, *]](
+def retryingOnSomeErrors[M[_]: Monad: Temporal, A, E: Handle[M, *]](
```
Comment (AlejandroBudy) [inline modules/docs/src/main/mdoc/docs/mtl-combinators.md:53]:
```suggestion
def retryingOnSomeErrors[M[_]: Temporal, A, E: Handle[M, *]](
```

File: modules/docs/src/main/mdoc/docs/mtl-combinators.md [hunk 3]
```diff
@@ -102,19 +103,20 @@ retry on all errors.
...
```
Comment (AlejandroBudy) [inline modules/docs/src/main/mdoc/docs/mtl-combinators.md:106]:
```suggestion
def retryingOnSomeErrors[M[_]: Temporal, A, E: Handle[M, *]](
```

File: modules/core/shared/src/main/scala/retry/package.scala [hunk 4]
```diff
@@ -146,35 +142,32 @@ package object retry:
...
     yield result
 
     wasSuccessful(a).ifM(
-      M.pure(Right(a)), // stop the recursion
+      T.pure(Right(a)), // stop the recursion
       onFalse
     )
 
-  private def retryingOnSomeErrorsImpl[M[_], A, E](
+  private def retryingOnSomeErrorsImpl[M[_], A, E <: Throwable](
```
Comment (cb372) [inline modules/core/shared/src/main/scala/retry/package.scala:163]: Could remove the `E` type param here and hardcode it to `Throwable`? It's `Throwable` at all the call sites.

File: modules/core/shared/src/test/scala/retry/SyntaxSuite.scala [hunk 5]
```diff
@@ -1,211 +1,212 @@
...
+      }
+
+    def getState: IO[State]  = stateRef.get
+    def getAttempts: IO[Int] = getState.map(_.attempts)
+
+  private val mkFixture: IO[Fixture] = Ref.of[IO, State](State()).map(new Fixture(_))
+
+  test("retryingOnFailures - retry until the action succeeds") {
+
+    val policy: RetryPolicy[IO]                 = RetryPolicies.constantDelay[IO](1.second)
```
Comment (AlejandroBudy) [inline modules/core/shared/src/test/scala/retry/SyntaxSuite.scala:45]: I can update the tests to 1 millisecond instead

File: modules/docs/src/main/mdoc/docs/combinators.md [hunk 6]
```diff
@@ -10,38 +10,38 @@ retries.
...
-| Combinator | Context bound | Handles |
-| --- | --- | --- |
-| [`retryingOnFailures`](#retryingOnFailures) | Monad | Failures |
-| [`retryingOnSomeErrors`](#retryingonsomeerrors) | MonadError | Errors |
-| [`retryingOnAllErrors`](#retryingonallerrors) | MonadError | Errors |
-| [`retryingOnFailuresAndSomeErrors`](#retryingonfailuresandsomeerrors) | MonadError | Failures and errors |
-| [`retryingOnFailuresAndAllErrors`](#retryingonfailuresandallerrors) | MonadError | Failures and errors |
+| Combinator                                                            | Context bound | Handles             |
+| --------------------------------------------------------------------- | ------------- | ------------------- |
+| [`retryingOnFailures`](#retryingOnFailures)                           | Monad         | Failures            |
```
Comment (cb372) [inline modules/docs/src/main/mdoc/docs/combinators.md:15]: A lot of this documentation will need updating, but let's leave it for now and do one big doc update at the end. We have a ticket for that.

**Review by ccantarero91 — APPROVED** [review]
LGTM

File: modules/core/shared/src/main/scala/retry/package.scala (hunk 4 above)
Comment (AlejandroBudy) [reply modules/core/shared/src/main/scala/retry/package.scala]: Removed!

**Review by cb372 — APPROVED** [review]
🚀

**Pull request merged**